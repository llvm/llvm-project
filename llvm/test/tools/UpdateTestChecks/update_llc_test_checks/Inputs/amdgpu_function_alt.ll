; RUN: cat %S/amdgpu_function_alt.s | FileCheck --check-prefixes=CHECK %s

define float @sample(float %x) {
  %y = fmul float %x, %x
  ret float %y
}
