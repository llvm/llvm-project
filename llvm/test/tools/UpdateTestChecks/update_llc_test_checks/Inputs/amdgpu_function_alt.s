	.text
	.amdgcn_target "amdgcn--amdpal--gfx1030"
	.globl	sample
	.p2align	2
	.type	sample,@function
sample:
	s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
	v_mul_f32_e32 v0, v0, v0
	s_setpc_b64 s[30:31]
.Lfunc_end0:
	.size	sample, .Lfunc_end0-sample

	.p2alignl 6, 3214868480
	.fill 48, 4, 3214868480
	.section	".note.GNU-stack","",@progbits
	.amd_amdgpu_isa "amdgcn--amdpal--gfx1030"
	.amdgpu_pal_metadata
---
amdpal.pipelines:
  - .api:            !str ''
    .shader_functions:
      sample:
        .backend_stack_size: 0
        .lds_size:       0
        .sgpr_count:     0x20
        .stack_frame_size_in_bytes: 0
        .vgpr_count:     0x1
amdpal.version:
  - 0x3
  - 0
...
	.end_amdgpu_pal_metadata
