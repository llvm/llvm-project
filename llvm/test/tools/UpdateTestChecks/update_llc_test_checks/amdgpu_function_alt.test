## Checks two things:
# - matching functions in AMDGPU assembly output without certain comments in the assembly
# - using the --tool and --default-march arguments

# Using `cat` as a tool requires some unusual setup (compared to the other test
# of UpdateTestChecks tools), because update_llc_test_checks only reliably
# substitutes %S (source directory) when evaluating RUN lines in the .ll file.
# The .ll file used during the test is a temporary copy, so we also need to
# provide a temporary copy of the cat'd .s file to avoid relying on
# implementation details of how llvm-lit chooses %t.

# RUN: rm -rf %t.dir && mkdir %t.dir
# RUN: cp -f %S/Inputs/amdgpu_function_alt.s %t.dir/
# RUN: cp -f %S/Inputs/amdgpu_function_alt.ll %t.dir/

# RUN: %update_llc_test_checks --llc-binary "" --tool cat --default-march=amdgcn --version 4 %t.dir/amdgpu_function_alt.ll
# RUN: diff -u %S/Inputs/amdgpu_function_alt.ll.expected %t.dir/amdgpu_function_alt.ll

# Check that re-running remembers the arguments and doesn't change anything

# RUN: %update_llc_test_checks --llc-binary "" %t.dir/amdgpu_function_alt.ll
# RUN: diff -u %S/Inputs/amdgpu_function_alt.ll.expected %t.dir/amdgpu_function_alt.ll
