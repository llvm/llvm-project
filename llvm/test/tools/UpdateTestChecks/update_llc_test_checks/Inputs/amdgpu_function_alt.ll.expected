; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --tool cat --default-march amdgcn --version 4
; RUN: cat %S/amdgpu_function_alt.s | FileCheck --check-prefixes=CHECK %s

define float @sample(float %x) {
; CHECK-LABEL: sample:
; CHECK:         s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
; CHECK-NEXT:    v_mul_f32_e32 v0, v0, v0
; CHECK-NEXT:    s_setpc_b64 s[30:31]
  %y = fmul float %x, %x
  ret float %y
}
